### PR TITLE
Improve UMA module and outcome handling

### PR DESCRIPTION
**Pull request by pemulis**
Signed-off-by: John Shutt <pemulis@users.noreply.github.com>

This PR resolves three outstanding issues:

1. Adds a button for the user to confirm that the proposal passed before showing a button to request execution.
2. Adds a button to delete a rejected proposal.
3. Adds handling for the case where there are no transactions in the proposal.

Notes:

- We have not been able to test the button to delete a rejected proposal since the Goerli Finder is pointing to the actual voting contract, rather than the mock oracle, for UMA V2 testing. We should be able to test this next week. In the meantime, a code review should be adequate. This button only exists to handle the edge case of a bad proposal that was not deleted during the dispute period, which is later duplicated by a good proposal, which requires you to delete the bad proposal (with the identical hash) before making the good proposal.
- We still have bad UX when a proposal has been executed and then a duplicate proposal is made in the future. Our code detects the earlier execution and does not present a button to propose again. Our immediate plan is to note this edge case in documentation, and direct users to make the proposal manually through their UMA Optimistic Governor interface in Zodiac. Our longer-term (2 month) plan is to add a `ProposalExecuted` event with a timestamp to the next iteration of the `OptimisticGovernor` contract, which will use the `OptimisticArbitrator` rather than the `OptimisticOracle`. We will encourage all users to migrate to `OptimisticGovernorV2`, anyway, and the migration process is relatively easy since Zodiac modules are...modular, and you can even have two Zodiac modules running side by side.

### CODE REVIEW

**Review by abg4 — APPROVED** [review]

File: src/plugins/safeSnap/utils/umaModule.ts [hunk 1]
```diff
@@ -108,6 +109,23 @@ export const getModuleDetailsUma = async (
       ]
     );
     timestamp = await moduleContract.proposalHashes(proposalHash);
+  } else {
```
Comment (mrice32) [inline src/plugins/safeSnap/utils/umaModule.ts:112]: Unrelated to the goal of this PR, but I think we should go back and re-examine the logic flow of this method. I think the logic/branching may be a bit more convoluted than it needs to be, and there may be error states that aren't handled gracefully (especially related to the transactions === undefined case),

**Review by mrice32 — APPROVED** [review]
LGTM!
